view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js

```diff
@@ -182,9 +182,16 @@ define(
 
             renderCheckoutComponent: function() {
                 this.isPlaceOrderAllowed(false);
-                let configuration = this.buildComponentConfiguration(this.paymentMethod(), this.paymentMethodsExtraInfo());
-
-                this.mountPaymentMethodComponent(this.paymentMethod(), configuration);
+                const configuration = this.buildComponentConfiguration(
+                    this.paymentMethod(),
+                    this.paymentMethodsExtraInfo()
+                );
+                this.mountPaymentMethodComponent(this.paymentMethod(), configuration).then(
+                    function() {
+                        const hasBillingAddress =  quote.billingAddress() !== null
+                        this.isPlaceOrderAllowed(hasBillingAddress);
+                    }
```

Comment:
I ran across this issue while debugging why `isPlaceOrderActionAllowed` was false for a non-Adyen payment method and I tried your solution.

I'm getting `Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'isPlaceOrderAllowed')`.

Switching the callback to an arrow function retained the proper `this` context and resolved the TypeError.

```suggestion
                    () => {
                        const hasBillingAddress =  quote.billingAddress() !== null
                        this.isPlaceOrderAllowed(hasBillingAddress);
                    }
```